internal/evidence/reactor.go

```diff
@@ -120,25 +120,19 @@ func (r *Reactor) handleEvidenceMessage(envelope *p2p.Envelope) error {
 	logger := r.logger.With("peer", envelope.From)
 
 	switch msg := envelope.Message.(type) {
-	case *tmproto.EvidenceList:
-		// TODO: Refactor the Evidence type to not contain a list since we only ever
-		// send and receive one piece of evidence at a time. Or potentially consider
-		// batching evidence.
-		//
-		// see: https://github.com/tendermint/tendermint/issues/4729
-		for i := 0; i < len(msg.Evidence); i++ {
-			ev, err := types.EvidenceFromProto(&msg.Evidence[i])
-			if err != nil {
-				logger.Error("failed to convert evidence", "err", err)
-				continue
-			}
-
-			if err := r.evpool.AddEvidence(ev); err != nil {
-				// If we're given invalid evidence by the peer, notify the router that
-				// we should remove this peer by returning an error.
-				if _, ok := err.(*types.ErrInvalidEvidence); ok {
-					return err
-				}
+	case *tmproto.Evidence:
+		// Process the evidence received from a peer
```

Comment:
Does this imply we could also (separately) get rid of the EvidenceList type? I see that we still use it in the Pool and a bunch of tests, but the only place we _obviously_ use muliple entries is in the test.

Reply:
I tried this and things went boom :) Right now, it is mainly used in evidence pool. In addition to the tests, the Block structure also has an EvidenceList within it. From a theoretical point of view, it might make sense to keep it as such if we have more than 1 evidence committed within a block. Do you know whether we indeed always commit only one evidence? How is this field populated (my guess is its populated from the evidence pool)?

Reply:
I don't know the answer either. I just did a quick skim of other uses and the only place I saw where we obviously populate multiple entries was in the reactor test. This is not something we need to investigate right now.

Reply:
@cmwaters I think your comment below is related to this. Right now the block stores a list of evidences and the nesting occurs because of this. Maybe we should investigate whether this should be removed as a resolution to the other issue. The naming is adding to the confusion (evidence list is named Evidence and then the evidence within the list is also named evidence).